ui/src/app/utils/cluster-initializer.spec.tsx

```diff
@@ -0,0 +1,305 @@
+import * as React from 'react';
+
+import {notebooksClusterApi, registerApiClient as registerApiClientNotebooks} from 'app/services/notebooks-swagger-fetch-clients';
+import {clusterApi, registerApiClient} from 'app/services/swagger-fetch-clients';
+import {ClusterInitializer} from 'app/utils/cluster-initializer';
+import {ClusterInitializerOptions} from 'app/utils/cluster-initializer';
+import {Cluster} from 'generated/fetch';
+import {ClusterStatus} from 'generated/fetch';
+import {ClusterApi} from 'generated/fetch/api';
+import SpyInstance = jest.SpyInstance;
+import expect = jest.Expect;
+import {ClusterApi as NotebooksClusterApi} from 'notebooks-generated/fetch';
+import {ClusterApiStub} from 'testing/stubs/cluster-api-stub';
+import {NotebooksClusterApiStub} from 'testing/stubs/notebooks-cluster-api-stub';
+
+let mockGetCluster: SpyInstance;
+let mockCreateCluster: SpyInstance;
+let mockDeleteCluster: SpyInstance;
+let mockStartCluster: SpyInstance;
+
+const baseCluster: Cluster = {
+  clusterName: 'aou-rw-3',
+  clusterNamespace: 'aou-rw-12345',
+  status: ClusterStatus.Running
+};
+
+describe('ClusterInitializer', () => {
+
+  beforeEach(() => {
+    jest.useFakeTimers();
+
+    registerApiClient(ClusterApi, new ClusterApiStub());
+    registerApiClientNotebooks(NotebooksClusterApi, new NotebooksClusterApiStub());
+
+    mockGetCluster = jest.spyOn(clusterApi(), 'getCluster');
+    mockCreateCluster = jest.spyOn(clusterApi(), 'createCluster');
+    mockDeleteCluster = jest.spyOn(clusterApi(), 'deleteCluster');
+    mockStartCluster = jest.spyOn(notebooksClusterApi(), 'startCluster');
+  });
+
+  afterEach(() => {
+    jest.useRealTimers();
+    jest.clearAllMocks();
+  });
+
+  const mockGetClusterCalls = (responses: Array<Cluster>) => {
```

Comment:
It would be simpler to have a version that takes in an array of statuses, since the base cluster props seem to be used everywhere.